lib/bamboo/adapters/smtp_adapter.ex

```diff
@@ -215,7 +215,19 @@ defmodule Bamboo.SMTPAdapter do
     |> add_smtp_line(text_body)
   end
 
-  defp add_attachment_header(body, %{content_type: content_type} = attachment)
+  defp add_attachment_header(body, attachment) do
+    case attachment.content_id do
```

Comment:
```suggestion
    case attachment[:content_id] do
```

So it won't fail when the key doesn't exist.

Reply:
I can do that but I don't see the scenario where this key doesn't exist.

`attachment` is a struct of type `Bamboo.attachment`. As `content_id` is defined in is `defstruct`, even if the developer didn't give a value when initializing  the struct, the key should be here. Right? 🤔

Reply:
Should we enforce the usage of this struct then? Because we are not doing pattern matching with it 🤔

Reply:
Oh I didn't see it was a private function, my bad

Reply:
It also depends on the versions of Bamboo we allow. 
In the past this field wasn't present (if I remember properly). Can you make sure the field was already present in the oldest Bamboo version we accept?

Reply:
That's a good point.

We ask for Bamboo `~>1.6` and this field was added in `1.4`.

Seems that we're good. 🙂